UF-Unionbuilder/LaborEnabler
# Fe preuser

This pull request provides the functionality for canvassers to create pre-user objects and have a printable way of accessing these slips.  This pull request also provides the functionality for users to sign up using their given association key.

To test:

1.  Login to the front end client using an account with Admin or Organizer Role.
2.  Click the Create Preuser button.
3.  Choose a Group Id.  If an invalid group id is chosen, then the request should fail.
4.  Choose the desired number of association keys to create.  Note that these will all be tied to the same group.
5.  Copy these keys down for future reference
6.  Log out / Remove Cookies from browser.
7.  Proceed back to the login page.  Click the Sign Up button.
8.  Enter the provided association key.  If the key is made up or has already been used, the request will fail.
9.  Enter the desired username.  If the username is missing or already taken, the request will fail.
10.  Enter the desired password.  If the password is missing, the request will fail.
